tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,485 @@
+        }
+
+        private static void DownloadAndExtractJitBenchRepo()
+        {
+            using (var client = new HttpClient())
+            {
+                // Download the JitBench repository and extract it.
+                const string repoUrl = "https://github.com/aspnet/JitBench";
+                const string sha1Id = "a44cd96e4ff8d7c1b43fe3c96a597919bb1544bd";
+                var archiveName = $"{sha1Id}.zip";
+                var url = $"{repoUrl}/archive/{archiveName}";
+                var zipFile = Path.Combine(s_temporaryDirectory, archiveName);
+
+                using (FileStream tmpzip = File.Create(zipFile))
+                {
+                    using (Stream stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+                Directory.Move(Path.Combine(s_temporaryDirectory, $"JitBench-{sha1Id}"), s_jitBenchDevDirectory);
+            }
+        }
+
+        private static IDictionary<string, string> InstallSharedRuntime()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture x64"
+            };
+            LaunchProcess(psi, 180000);
+
+            // TODO: This is currently hardcoded, but we could probably pull it from the powershell cmdlet call.
+            return new Dictionary<string, string> { { "PATH", $"{Path.Combine(s_jitBenchDevDirectory, ".dotnet")};{psi.Environment["PATH"]}" } };
+        }
+
+        private static void InstallDotnet()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -InstallDir .dotnet -Channel master -Architecture x64"
+            };
+            LaunchProcess(psi, 180000);
+        }
+
+        private static void ModifySharedFramework()
+        {
+            // Current working directory is the <coreclr repo root>/sandbox directory.
+            Console.WriteLine($"Modifying the shared framework.");
+
+            var sourcedi = new DirectoryInfo(Directory.GetCurrentDirectory());
+            var targetdi = new DirectoryInfo(
+                new DirectoryInfo(Path.Combine(s_jitBenchDevDirectory, ".dotnet", "shared", "Microsoft.NETCore.App"))
+                .GetDirectories("2.1*").First().FullName);
```

Comment:
Currently, this harness downloads a fixed version of JitBench, so the version would not move behind the scenes, and I agree we should port this to the release/2.0.0 branch.